app/models/smart_proxy_feature.rb

```diff
@@ -0,0 +1,30 @@
+class SmartProxyFeature < ApplicationRecord
+  belongs_to :smart_proxy
+  belongs_to :feature
+  validates :feature, :uniqueness => {:scope => :smart_proxy_id}
+
+  serialize :capabilities, Array
+  store :settings, coder: JSON
+
+  def details
+    {settings: settings, capabilities: capabilities}
+  end
+
+  def self.import_features(smart_proxy, features_json)
+    name_map = Feature.name_map
+    new_feature_classes = features_json.keys.map {|feature| name_map[feature]}
+    smart_proxy.smart_proxy_features.where.not(feature_id: new_feature_classes).destroy_all
```

Comment:
should this be either `feature: new_feature_classes` or `feature_id: new_feature_classes.map(&:id)`? (i'm guessing rails handles it anyways, but just for clarity)